platform-sdk/swirlds-platform-core/src/jmh/java/com/swirlds/platform/core/jmh/EventBenchmarks.java

```diff
@@ -96,8 +96,8 @@ public void serializeDeserialize(final Blackhole bh) throws IOException {
         //
         // Benchmark                                (seed)   Mode  Cnt    Score    Error   Units
         // EventSerialization.serializeDeserialize       0  thrpt    3  962.486 ± 29.252  ops/ms
-        EventSerializationUtils.serializePlatformEvent(outStream, event, true);
-        bh.consume(EventSerializationUtils.deserializePlatformEvent(inStream, true));
+        outStream.writePbjRecord(event.getGossipEvent(), GossipEvent.PROTOBUF);
```

Comment:
Note that the `writePbjRecord` API is currently error-prone if someone passes an incorrect Codec for a given record. PBJ records have a public static final PROTOBUF field that refers to their codec, but given that the `writePbjRecord` accepts a `T extends Record`, one would have to use reflection to fetch the PROTOBUF value.

I understand that using Java reflection may be undesirable (for performance reasons, or complexity), but at the same time this makes it easy to introduce a bug by supplying an incorrect codec to the API.

Any thoughts on this?

Reply:
the method `writePbjRecord()` uses generics to indicate that the instance and codec should be the same type. I would consider this to be sufficient. But if we want to make it safer and simpler, I don't think reflections is the way to go. We should make all PBJ records implement an interface with a method `getProtoCodec()`, that way you get the codec directly from the instance you are serializing.

Reply:
Got it. I think that the type constraints imposed by the generic is a good first step to prevent most of the compile-time errors.

But I'd really like to see a runtime protection as well which may be important when writing a generic code that would use this method to write arbitrary records. I totally like the idea of an interface with a `getProtoCodec()` method! However, I suppose that adding it is outside of scope of this particular PR. I'd appreciate it if you could file an issue against PBJ to implement this interface.

Reply:
https://github.com/hashgraph/pbj/issues/287